exf/deployment/scheduler/ciaox11_exf_dispatcher.h

```diff
@@ -88,26 +91,50 @@ namespace CIAOX11
           /// Allocate this instance for executing a next task
           /// @retval true Instance has been allocated for a next task
           /// @retval false Instance was not allocated, already busy with a task
+          /// @note This operation must be called from the dispatcher with
+          /// its _g_queue lock held, this operation is *NOT* thread safe
+          /// in itself
           bool allocate ()
-          { return !this->busy_.exchange(true); }
+          {
+            // When have a concurrent flag greater as zero we can dispatch
+            // a task to it
+            if (this->concurrent_ > 0)
```

Comment:
possible race condition

Reply:
Thanks for checking, correct that there is no locking here now, locking is now done at the dispatcher level, there I now lock, at the moment the instance is released by the dispatcher there is also a lock there. An atomic doesn't work anymore, so the other option is to add a mutex here, which means we do a double lock.

Reply:
Renamed the operation to `allocate_i`, already has a note about the locking requirements